lxd/main_init.go

```diff
@@ -494,8 +510,148 @@ they otherwise would.
 			return err
 		}
 	}
+	return nil
+}
+
+// Run the logic for preseed mode
+func (cmd *CmdInit) runPreseed(client lxd.ContainerServer) error {
+	data := &cmdInitData{}
+
+	err := cmd.Context.InputYAML(data)
+	if err != nil {
+		return fmt.Errorf("Invalid preseed YAML content")
+	}
+
+	return cmd.run(client, data)
+}
+
+// Apply the configuration specified in the given init data.
+func (cmd *CmdInit) run(client lxd.ContainerServer, data *cmdInitData) error {
```

Comment:
On a second thought, I actually realize that we must have already APIs to fully revert things like storage or network bridge creation. So why do you say above it's not possible to have this function fully transactional? Because the "revert" logic could itself fail at some random point? At least in principle.